WeatherApp/WeatherApp/App/Data/Repositories/WeatherRepository.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+protocol WeatherRepositoryProtocol {
+
+    func fetchWeather(for cityName: String, completion: @escaping (Result<WeatherModel, ClientError>) -> Void)
+
+}
+
+class WeatherRepository: WeatherRepositoryProtocol {
+
+    let weatherService: WeatherServiceProtocol
+
+    init(weatherService: WeatherServiceProtocol) {
+        self.weatherService = weatherService
+    }
+
+    func fetchWeather(
+        for cityName: String,
+        completion: @escaping (Result<WeatherModel, ClientError>) -> Void
+    ) {
+        weatherService.fetchWeather(for: cityName) { result in
+            switch result {
+            case .success(let currentWeatherResponse):
+                let weatherModel = self.mapToWeatherModel(response: currentWeatherResponse)
+                completion(.success(weatherModel))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    private func mapToWeatherModel(response: CurrentWeatherResponse) -> WeatherModel {
+        let weatherDescription = response.weather.first?.description ?? "Not Avaliable"
+
+        return WeatherModel(
```

Comment:
I believe you already did this somewhere, but create a new init for `WeatherModel`. If you have done this already just leave a comment you did

Reply:
I did it already on previous PR.